### ISSUE TITLE
[DateRangePicker] Ability to select one date only at a time (start or end)

### ISSUE THREAD

**Issue by Rupinderthind** [mui/mui-x#4624]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

For now when we select start date only and trying to close datepicker then it getting reset to previous dates, we have to select both start and end date everytime.

### Expected behavior 🤔

It should allow to select one only at a time, means if I select start date only it should not required to have end date as well.

### Steps to reproduce 🕹

Steps:

1. click on start date input box
2. select start date
3. then click outside to close datepicker
4. dates will be reset to previous

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by Rupinderthind** [comment]
duplicate issue

**Comment by BjoernRave** [comment]
I can confirm this issue

**Comment by alexfauquette** [comment]
If I correctly understand, the idea would be to add a prop `allowPartialSelection` on `DateRangePickers`

**Comment by flaviendelangle** [comment]
Doing the reproduction steps, this bug does not exist anymore.
Feel free to comment if I'm missing something

Maybe it has been fixed by #4408
